### DISCUSSION TITLE
🐞 14.0.4 broke everything again + the minimal repo requirement for issues is silly

### DISCUSSION THREAD

**Discussion by oagix** [vercel/next.js#59534] [Help]
### Summary

There's so many cases where it's nearly impossible to provide a minimal repro repo. Because:
1. Complexity
2. Private Code
3. Iteration Velocity

After upgrading to 14.0.4, imports are starting to break again:

![image](https://github.com/vercel/next.js/assets/8274669/645247e8-6042-4909-9fc7-a5044ad5e151)

And there's people in the community having the same issue, but the issue bot is dimissing because of no repro:

https://github.com/vercel/next.js/issues/59386

Would love to see a better system to handle these situations, as well as an explanation around why minor version of Next.js tend to keep breaking imports. Happened on 14.0.1 or 2 I believe, 3 was the first one that worked, but now 4/5-canary aren't working again.

**Comment by oagix** [comment]
This is happening when importing the `Grid` component from `@mui/material`, but all the other ones work.

**Comment by timneutkens** [comment]
Unfortunately we can't track down what is causing the issue you're running into in particular based on the  screenshot. I tried to create a reproduction based on your message but wasn't able to: https://github.com/timneutkens/mui-test.

> as well as an explanation around why minor version of Next.js tend to keep breaking imports. Happened on 14.0.1 or 2 I believe, 3 was the first one that worked, but now 4/5-canary aren't working again.

It highly depends on the library that you're using, which is likely not covered in the Next.js test suite and can have i.e. incorrect ESM configuration or other overrides. I.e. using `module` vs `main` and such. It's hard to say based on the range of versions given, if you're able to narrow it down to a particular canary version that would be helpful if you can't share any code.

**Reply by raajnadar** [reply]
I can confirm the problem is between `14.0.3` to `14.0.4` upgrade,

My initial version was `14.0.0` directly moving to `14.0.4` gave the error, to get the error I nuked the `node_modules` as well as the `build` folder of next js.

**Comment by phibo23** [comment]
We have observed the same error occuring after upgrading from 14.0.3 to 14.0.4, also with mui/material in server components.
I was able to narrow our specific problem down to several components where we use mui `SvgIcon`. After explicitely adding `use client` in those components, the pages could be rendered again when requested directly, but the error still occured when linking to those pages with `NextLink`.

Unfortunately, I also wasn't able to reproduce the error in a minimal example until now. I'll keep trying.

I don't think it's imports but rendering of certain components in `Suspense`

**Comment by RoudyTarabay** [comment]
Happening also with Antd. It seems it's happening due to importing the same package in multiple components used within the page.tsx file.

**Comment by ghost** [comment]
@timneutkens I created a reproduction repository: https://github.com/schimi-dev/next-mui-bug-reproduction and a corresponding issue https://github.com/vercel/next.js/issues/59804.

**Comment by al222cv** [comment]
Are you guys using CssBaseline? I found out that if I import CssBaseline like this: 

`import CssBaseline from '@mui/material/CssBaseline'`

instead of 

`import { CssBaseline } from '@mui/material'`

makes the error disappear.

**Reply by raajnadar** [reply]
I already have the same code but still getting the error

I am doing this "import CssBaseline from '@mui/material/CssBaseline'"

**Comment by bjfresh** [comment]
Have put together a new issue + minimal reproduction that will hopefully help get this problem fixed:

https://github.com/vercel/next.js/issues/60449

**Comment by wiiiimm** [comment]
Just writing to inform everyone that this issue has been fixed in the latest Next.js canary release [v14.0.5-canary.62](https://github.com/vercel/next.js/releases/tag/v14.0.5-canary.62) .

**Reply by wiiiimm** [reply]
The patch has been shipped officially under [Next.js v14.1](https://github.com/vercel/next.js/releases/tag/v14.1.0).

**Reply by diogowommer** [reply]
I had the same problem when I used theme with antd. Problem solved with Next v14.1. 
Thanks!

**Comment by donatcranburi** [comment]
also ran into this I think, manifested it self as `.send is not a function` when called on an instance of the the S3Client class from `@aws-sdk/client-s3`

eg:
```
TypeError: A.send is not a function
  at a (/foo/bar/.next/server/chunks/877.js:1:4087)
  at x (/foo/bar/.next/server/app/page.js:1:10498)
  at u (/foo/bar/.next/server/app/page.js:1:5771)
```

Was building fine at 14.0.2, and 14.0.3, throwing errors at 14.0.4, fixed at 14.0.5-canary.62 as mentioned above. Mostly posting here in case it helps anyone on google.

**Reply by donatcranburi** [reply]
hmm I think this was my bad. I had `"use server"` at the top of a file (instead of at the top of an async function) that exported a named (syncronous/not-async) function, and it seems like maybe `"use server"` was causing the function export to be wrapped in a promise at build time (and `.send` is not a method on the promise). Odd that it built fine on the 14.0.2 and 14.0.5-canary.62 versions, broke again when I got up to updating the minor versions (14.2.5) though. documenting for posterity.
